gc/base/MarkingScheme.cpp

```diff
@@ -392,6 +393,8 @@ MM_MarkingScheme::createWorkPackets(MM_EnvironmentBase *env)
 #if defined OMR_GC_MODRON_CONCURRENT_MARK
 		workPackets = MM_WorkPacketsConcurrent::newInstance(env);
 #endif /* defined OMR_GC_MODRON_CONCURRENT_MARK */
+	} else if (_extensions->isSnapshotAtTheBeginningBarrierEnabled()) {
+		workPackets = MM_WorkPacketsSATBBarrier::newInstance(env);
```

Comment:
Something about ext->concurrentMark flag should change. Either, it should should be renamed to something incrementalUpdateBarrier is enabled or keep it true even when SATB barrier is enabled (in which case isSATBBarrierEnebled flag could be upgraded to writeBarrierMode enum, which would then make this is similar code have another else branch asserting the mode is one of the two possible).

Dont have a strong opinion which of the two approaches (let's see if anybody else has an opinion), but bottom line concurrentMark has fairly generic name and somewhat implies it could mean SATB being enabled is also a posiblity.

I understand this would affect not only but a few other places in the code (including OpenJ9)

Reply:
I agree with @amicic that something should be done here.  My initial thought is that we leave the boolean as is and if concurrentMark is enabled then use the writebarriertype to decide which of the two WorkPacket structures to create.  I assume that pattern would get used in a few place.

Reply:
I moved isSnapshotAtTheBeginningBarrierEnabled() to MM_Configuration and modified it to check for _writeBarrierType which is set when configuration is instantiated.